### PR TITLE
Refactor Database

### PR DESCRIPTION
**Pull request by stdevAdrianPaez**
## Changes:
- Refactoring the database implementation

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by D-DePablos** [comment]
For the sake of completeness, everything works locally for me, and the db was able to continue sync with previously created state (had ~60 blocks), so sync continued correctly.

File: docs/docs/features/database.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+type DatabaseTxOp func(txn DatabaseOperations) error
+```
+
+**RunTXn** receives a function that has a unique param that can do all the database operations, those operations are
+committed only if the `DatabaseTxOp` function ends without error.
+
+# MDBX Database
+
+Now **Juno** uses an MDB database to store all its data. An MDB database is an environment (a database file) that can
```
Comment (stdevMac) [inline docs/docs/features/database.mdx:60]: MDB => MDBX

**Review by stdevMac — APPROVED** [review]

File: docs/docs/features/database.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+title: Database
+---
+
+# General characteristics
+
+**Juno** defines a database as an object that can store **key-value** pairs in a persistent way. Keys and values are
```
Comment (joshklop) [inline docs/docs/features/database.mdx:7]: I think it would be slightly easier to read if "Juno" always had a normal (non-bold) font face.

**Review by joshklop — COMMENTED** [review]
This looks like awesome work to me. Thank you!

File: cmd/juno/cli/root.go [hunk 3]
```diff
@@ -61,15 +61,17 @@ var (
...
 			}
 
 			if config.Runtime.Metrics.Enabled {
 				s := metric.SetupMetric(":" + strconv.Itoa(config.Runtime.Metrics.Port))
+				// Initialize the Metrics Service.
 				processHandler.Add("Metrics", s.ListenAndServe, s.Close)
 			}
 
-			if err := db.InitializeDatabaseEnv(config.Runtime.DbPath, 10, 0); err != nil {
+			if err := db.InitializeMDBXEnv(config.Runtime.DbPath, 100, 0); err != nil {
```
[the remark this answers is not in the document]
Comment (stdevAdrianPaez) [reply cmd/juno/cli/root.go:74]: It is the maximum number of named databases that we can have in the MDBX environment, maybe we can have it in the configuration file

**Review by tshakalekholoane — APPROVED** [review]
LGTM. Excellent work!

**Review by joshklop — APPROVED** [review]

**Pull request merged**